### PR TITLE
fix(timestamps): account for timestamps of 0 when creating Dates

### PR DESCRIPTION
**Pull request by ImRodry**
**Please describe the changes this PR makes and why it should be merged:**
For some reason, @kyranet's suggestion in #7218 was ignored so this PR fixes all instances where timestamps could be 0 on the library. I had to disable the eqeqeq rule for this in order to check for null and undefined timestamps but if you have a better idea please let me know.

Note about the change on the GuildMemberManager file: Discord throws an error when passing a timestamp that is more than 28 days in the past or the future, however, without this fix you'd be able to pass 0 and get no error, as the API would treat that as null which ends up doing the same thing but is not exactly the expected behavior. With this fix, we transform every timestamp into an ISO string and let Discord error when it should. I'll gladly undo this if you think it's necessary however

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

File: packages/discord.js/src/structures/MessageEmbed.js [hunk 1]
```diff
@@ -88,8 +88,8 @@ class MessageEmbed {
      * @type {?number}
      */
     // Date.parse() cannot be used here because data.timestamp might be a number
-    // Additionally, the nullish coalescing operator cannot be used here as we're checking for 0
-    this.timestamp = new Date(data.timestamp).getTime() || null;
+    // eslint-disable-next-line eqeqeq
+    this.timestamp = data.timestamp != null ? new Date(data.timestamp).getTime() : null;
```
Comment (imranbarbhuiya) [inline packages/discord.js/src/structures/MessageEmbed.js:92]: I'm just curious to know that why we can't use `!==` here?

Comment (ImRodry) [reply packages/discord.js/src/structures/MessageEmbed.js:92]: Read the description

Comment (imranbarbhuiya) [reply packages/discord.js/src/structures/MessageEmbed.js:92]: Oh sorry, got it's because of 0. right?

Comment (ImRodry) [reply packages/discord.js/src/structures/MessageEmbed.js:92]: No... it's because of undefined

**Comment by ImRodry** [comment]
Accidentally forgot to commit some changes, all good now

**Review by vladfrangu — APPROVED** [review]

File: packages/discord.js/src/managers/GuildMemberManager.js [hunk 2]
```diff
@@ -266,7 +266,10 @@ class GuildMemberManager extends CachedManager {
     _data.roles &&= _data.roles.map(role => (role instanceof Role ? role.id : role));
 
     _data.communication_disabled_until =
-      _data.communicationDisabledUntil && new Date(_data.communicationDisabledUntil).toISOString();
+      // eslint-disable-next-line eqeqeq
+      _data.communicationDisabledUntil != null
+        ? new Date(_data.communicationDisabledUntil).toISOString()
+        : _data.communicationDisabledUntil;
```
Comment (Jiralite) [inline packages/discord.js/src/managers/GuildMemberManager.js:272]: Why is this change needed? 0 is not a value that is accepted here?

**Review by thebigwolfy — APPROVED** [review]

File: packages/discord.js/src/managers/GuildMemberManager.js (hunk 2 above)
Comment (ImRodry) [reply packages/discord.js/src/managers/GuildMemberManager.js:272]: Please read the description of the PR

**Review by iCrawl — APPROVED** [review]

File: packages/discord.js/src/managers/GuildMemberManager.js (hunk 2 above)
Comment (NotSugden) [reply packages/discord.js/src/managers/GuildMemberManager.js:272]: would `_data.communicationDisabledUntil ??= ...` work here ?

Comment (ImRodry) [reply packages/discord.js/src/managers/GuildMemberManager.js:272]: That would only work if the value was nullish, and &&= wouldn't either because of 0

Comment (NotSugden) [reply packages/discord.js/src/managers/GuildMemberManager.js:272]: oh yes, im reading the logic in reverse MB

**Review by kyranet — APPROVED** [review]
